src/Compilers/CSharp/Portable/CodeGen/EmitExpression.cs

```diff
@@ -3762,29 +3762,26 @@ private void EmitConditionalOperator(BoundConditionalOperator expr, bool used)
 
             static bool hasIntegralValueZeroOrOne(BoundExpression expr, out bool isOne)
             {
-                if (expr.ConstantValueOpt is { } constantValue)
+                var constantValue = expr.ConstantValueOpt;
+                switch (constantValue)
```

Comment:
>switch (constantValue) [](http://example.com/codeflow?start=16&length=22)

It looks like this could be extracted to a `HasIntegralValueZeroOrOne(out bool isOne)` method on `ConstantValue`, usable from C# and VB.